### PR TITLE
Rewrite readme for unified, more examples

### PR DESCRIPTION
**Pull request by wooorm**
See rendered readme: https://github.com/remarkjs/react-markdown/tree/docs#readme

* Lowercase filenames to scream less
* Add `remark-cli`, `remark-preset-wooorm` to format prose
* Use imports and fancy new JS features
* Add link to a markdown cheatsheet, as many questions are about that
* Add details tags showing equivalent JSX for examples, to clarify how markdown works
* Refactor options list, clean signatures
* Add examples on plugins, plugins w/ options, renderers (syntax highlighting), and combining plugins w/ renderers (math)
* Add sections on why to use this, security, contributing, related projects
* Add test for `react-katex` integration

Closes GH-70.
Closes GH-191.
Related to GH-474.
Closes GH-479.
Closes GH-480.
Related to GH-483.

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -0,0 +1,502 @@
...
+
+## Why this one?
+
+There are other ways for markdown in React out there so why use this one?
+The two main reasons are that they often rely on `dangerouslySetInnerHTML` or
+have bugs with how they handle markdown.
+`react-markdown` uses a syntax tree to build the virtual dom which allows for
+updating only the changing DOM instead of completely overwriting.
+And `react-markdown` is 100% CommonMark (optionally GFM) compliant and has the
+syntax extensions and plugins you may want.
```
Comment (ChristianMurphy) [inline readme.md:32]: Hmmm :thinking: 
maybe:

```suggestion
`react-markdown` is 100% CommonMark (optionally GFM) compliant and has extensions to support custom syntax.
```
?

File: readme.md [hunk 2]
```diff
@@ -0,0 +1,502 @@
...
+
+## Examples
+
+### Use a plugin
+
+This example shows how to use a plugin.
+In this case, [`remark-gfm`][gfm], which adds support for which adds support for
+strikethrough, tables, tasklists and URLs directly:
+
+```js
```
Comment (ChristianMurphy) [inline readme.md:145]:
```suggestion
```jsx
```

File: readme.md [hunk 3]
```diff
@@ -0,0 +1,502 @@
...
+</details>
+
+### Use a plugin with options
+
+This example shows how to use a plugin and give it options.
+To do that, use an array with the plugin at the first place, and the options
+second.
+[`remark-gfm`][gfm] has an option to allow only double tildes for strikethrough:
+
+```js
```
Comment (ChristianMurphy) [inline readme.md:212]:
```suggestion
```jsx
```

**Review by ChristianMurphy — APPROVED** [review]

File: readme.md (hunk 1 above)
Comment (wooorm) [reply readme.md:32]:
```suggestion
`react-markdown` is 100% CommonMark (optionally GFM) compliant and has
extensions to support custom syntax.
```

**Comment by wooorm** [comment]
poor travis

**Review by ChristianMurphy — APPROVED** [review]

**Review by rexxars — APPROVED** [review]
👍

**Pull request merged**